pandas/_libs/index.pyx

```diff
@@ -355,7 +355,7 @@ cdef Py_ssize_t _bin_search(ndarray values, object val) except -1:
         Py_ssize_t mid = 0, lo = 0, hi = len(values) - 1
         object pval
 
-    if hi >= 0 and val > util.get_value_at(values, hi):
+    if hi == 0 or (hi > 0 and val > util.get_value_at(values, hi)):
```

Comment:
is there any user code that hits this? e.g. is there some example that failed before and now works? or just an untested corner case?

Reply:
the only caller is INdexEngine.get_loc, and Index constructor (I imagine all of them) will refuse to create an empty index. So, this cannot be triggered. Reviewers pressed, so I responded.